discovery/pkg/sync/endpoints_test.go

```diff
@@ -138,7 +141,166 @@ func TestUpdateEndpoints(t *testing.T) {
+		existingendpoints v1.Endpoints
+		expectErr         bool
+		expectedCount     float64
+		expectedTimestamp float64
+		expectedLabels    map[string]string
+	}{
+		{
+			name:       "add new endpoint resource",
+			actionKind: actionAdd,
+			endpoints: v1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "bar",
+					Labels: map[string]string{
+						"gimbal.heptio.com/backend": backendName,
+					},
+				},
+				Subsets: []v1.EndpointSubset{
+					{
+						Addresses: []v1.EndpointAddress{{IP: "192.168.0.1"}},
+						Ports:     []v1.EndpointPort{{Port: 80}},
+					},
+				},
+			},
+			existingendpoints: v1.Endpoints{},
+			expectedCount:     float64(1),
+			expectedTimestamp: 9.467208e+08,
+			expectedLabels: map[string]string{
+				"backendname": backendName,
+				"backendtype": backendType,
+				"namespace":   "foo",
+			},
+		},
+		{
+			name:       "add new endpoint resource, multiple ips",
+			actionKind: actionAdd,
+			endpoints: v1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "bar",
+					Labels: map[string]string{
+						"gimbal.heptio.com/backend": backendName,
+					},
+				},
+				Subsets: []v1.EndpointSubset{
+					{
+						Addresses: []v1.EndpointAddress{{IP: "192.168.0.1"}},
+						Ports:     []v1.EndpointPort{{Port: 80}},
+					},
+					{
+						Addresses: []v1.EndpointAddress{{IP: "192.168.0.2"}, {IP: "192.168.0.3"}},
+						Ports:     []v1.EndpointPort{{Port: 443}},
+					},
+				},
+			},
+			existingendpoints: v1.Endpoints{},
+			expectedCount:     float64(3),
+			expectedTimestamp: 9.467208e+08,
+			expectedLabels: map[string]string{
+				"backendname": backendName,
+				"backendtype": backendType,
+				"namespace":   "foo",
+			},
+		},
+		{
+			name:       "add new endpoint resource, existing-non gimbal",
+			actionKind: actionAdd,
+			endpoints: v1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "bar",
+					Labels: map[string]string{
+						"gimbal.heptio.com/backend": backendName,
+					},
+				},
+				Subsets: []v1.EndpointSubset{
+					{
+						Addresses: []v1.EndpointAddress{{IP: "192.168.0.1"}, {IP: "192.168.0.2"}},
+						Ports:     []v1.EndpointPort{{Port: 80}},
+					},
+				},
+			},
+			existingendpoints: v1.Endpoints{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "ep-existing",
+					Labels: map[string]string{
+						"who": "dis",
+					},
+				},
+				Subsets: []v1.EndpointSubset{
+					{
+						Addresses: []v1.EndpointAddress{{IP: "192.168.0.1"}},
+						Ports:     []v1.EndpointPort{{Port: 80}},
+					},
+				},
+			},
+			expectedCount:     float64(2),
+			expectedTimestamp: 9.467208e+08,
+			expectedLabels: map[string]string{
+				"backendname": backendName,
+				"backendtype": backendType,
+				"namespace":   "foo",
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			nowFunc = func() time.Time {
```

Comment:
How/where is this used?

Reply:
The Timestamp metric used `time.Now()` to get the current time. In order to unit test that metric, we need a way to set it in the test. If it's not set in the test, then it defaults to now(). Both the service & endpoint in package sync use this.